src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -4275,10 +4275,17 @@ internal static IEnumerable<CompletionResult> CompleteFilename(CompletionContext
                                                     continue;
 
                                                 var fileInfo = new FileInfo(entry);
-                                                if ((fileInfo.Attributes & FileAttributes.Hidden) != 0)
+                                                try
                                                 {
-                                                    PSObject wrapper = PSObject.AsPSObject(entry);
-                                                    psobjs.Add(wrapper);
+                                                    if ((fileInfo.Attributes & FileAttributes.Hidden) != 0)
+                                                    {
+                                                        PSObject wrapper = PSObject.AsPSObject(entry);
+                                                        psobjs.Add(wrapper);
+                                                    }
+                                                }
+                                                catch
+                                                {
+                                                    // do nothing if can't get file attributes
```

Comment:
We enumerate files with IgnoreInaccessible = true (line 4251) and this works on Windows but not Unix. It looks like .Net Core issue. Maybe fix there?

Reply:
The file is enumerated which isn't inaccessible at enumeration time, it fails when reading the file attributes

Reply:
I don't understand you comment. If the file is not enumerated it is not in the cycle. Really it is enumerated that converse  IgnoreInaccessible = true.

Reply:
Also the issue for WSL1. Can you check on WSL2?

Reply:
Looking a history of the issue I think the fix is well for completion.

Reply:
I tested on WSL2.  My comment is that the files that cause the exception are enumerated fine.  So it's not a problem with enumeration and IgnoreInaccessible is an EnumerationOption.  The problem is specifically when the file's attributes are being read.  So you have permission to see the file, but not anything about the file.